weave-js/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,65 @@
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+import React from 'react';
+import {twMerge} from 'tailwind-merge';
+
+import {Button} from '../Button';
+import {IconName} from '../Icon';
+import {Tailwind} from '../Tailwind';
+import {ToggleButtonGroupSize} from './types';
+
+export type ToggleButtonGroupProps = {
+  options: string[];
+  value: string;
+  size: ToggleButtonGroupSize;
+  isDisabled?: boolean;
+  icons?: Array<IconName | undefined>;
```

Comment:
maybe we should just create a new type `ToggleOption`
which includes the option label and the option icon. that way we don't have to do these two separate arrays and need to use the index to match them. instead if we create a type then the label + option become more tightly coupled

Reply:
```
type ToggleOption = {
  label: string;
  icon?: IconName
}

export type ToggleButtonGroupProps = {
  ...
  options: ToggleOption[] // label and icon are tightly coupled
}